### ISSUE TITLE
EXWM spinning, keyboard lag, not sure how to diagnose

### ISSUE THREAD

**Issue by doriantaylor** [ch11ng/exwm#912]
This is more of a meta-bug to the extent that I could use some help tracking down spinning in Emacs running EXWM. The load on my system is consistently hovering above 1 and `emacs` is taking up most of it, even when ostensibly idle. A document (or addendum to the README) describing how to profile Emacs running EXWM such that the conditions can be successfully replicated without being to intrusive would be helpful.

Thanks

**Comment by chookity-pokk** [comment]
Maybe try using `M-x:profiler-start` and then use it until it starts lagging then use `M-x:profiler-report`. That should give you a list of functions that are taking up the resources and how much they are using.

**Comment by medranocalvo** [comment]
Yes, `profiler` should show where the issue lies. Have you tried, did it help?  If so, mentioning in the FAQ would definitely help others.
